st2api/st2api/controllers/v1/actionexecutions.py

```diff
@@ -71,6 +73,20 @@ class ActionExecutionsControllerMixin(RestController):
         'trigger_instance'
     ]
 
+    def _get_from_model_kwargs_for_request(self, request):
+        from_model_kwargs = {'mask_secrets': cfg.CONF.api.mask_secrets}
+        query_string = request.query_string
+        query_params = dict(urlparse.parse_qsl(query_string))
+
+        show_secrets = query_params.get('show_secrets', 'false')
+        show_secrets = show_secrets.lower()
+        show_secrets = show_secrets == 'true'
+
+        if show_secrets and request_user_is_admin(request=request):
+            from_model_kwargs['mask_secrets'] = False
```

Comment:
So if a non-admin user asks for unmasked secrets, should be actually fail the request with BAD_REQUEST?